### PR TITLE
(PA-5409) Update Configure flags for openssl3

### PR DESCRIPTION
**Pull request by tvpartytonight**
Small update to the configure flags used with openssl 3.

### CODE REVIEW

**Comment by tvpartytonight** [comment]
I successfully test building with these flags [here](https://jenkins-platform.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2059/).

File: configs/components/openssl-3.0.rb [hunk 1]
```diff
@@ -133,13 +133,26 @@
...
+    'no-gost',
     target,
     sslflags,
     'no-camellia',
     'no-ec2m',
     'no-md2',
-    'no-ssl3'
+    'no-ssl3',
+    'no-ssl3-method',
+    'no-dtls1-method',
```
Comment (joshcooper) [inline configs/components/openssl-3.0.rb:144]:
```suggestion
    'no-dtls',
    'no-dtls1-method',
```

File: configs/components/openssl-3.0.rb [hunk 2]
```diff
@@ -133,13 +133,26 @@
...
+    'no-dtls1_2-method',
+    'no-aria',
+    'no-bf',
+    'no-cast',
```
Comment (joshcooper) [inline configs/components/openssl-3.0.rb:148]:
```suggestion
    'no-cast',
    'no-idea',
```

File: configs/components/openssl-3.0.rb [hunk 3]
```diff
@@ -133,13 +133,26 @@
...
+    'no-rc2',
+    'no-rc5',
```
Comment (joshcooper) [inline configs/components/openssl-3.0.rb:150]:
```suggestion
    'no-rc5',
    'no-seed',
```

File: configs/components/openssl-3.0.rb (hunk 1 above)
Comment (tvpartytonight) [reply configs/components/openssl-3.0.rb:144]: `no-dtls` was already specified [here](https://github.com/tvpartytonight/puppet-runtime/blob/PA-5409/configs/projects/_shared-agent-settings.rb#L229), so I wasn't sure if we wanted to repeat them?

Comment (tvpartytonight) [reply configs/components/openssl-3.0.rb:144]: I think it is the same for the other suggestions you have; they are already in the `shared-agent-settings.rb`.

Comment (joshcooper) [reply configs/components/openssl-3.0.rb:144]: ohhh, right I forgot about `openssl_extra_configure_flags`

**Review by joshcooper — APPROVED** [review]

**Pull request merged**